recipe-hooks-project/src/Register.js

```diff
@@ -0,0 +1,51 @@
+import React, {memo} from "react";
+import {Redirect} from "react-router-dom";
+import axios from "axios";
+import {Container, Form, FormTitle, Input, SubmitButton} from "./components/Form"
+import useInputState from "./hooks/useInputState";
+import useToggleState from "./hooks/useToggleState";
+
+function Register() {
+  const [email, handleEmailChange] = useInputState("");
+  const [password, handlePasswordChange] = useInputState("");
+  const [name, handleNameChange] = useInputState("");
+  const [redirect, toggleRedirect] = useToggleState(false)
+
+  async function handleRegistration(evt) {
+    evt.preventDefault();
+    try {
+      const response = await axios.post("http://localhost:8000/api/user/create/",
+        {email: email, name: name, password: password}
+      );
+      if (response.status === 201) {
+        console.log("User created!");
+        toggleRedirect();
+      }
+    } catch (e) {
+      console.log("Registration failed"); //FIXME message to user?
+    }
+  }
+
+  const renderRedirect = () => {
```

Comment:
I wouldn't put this as a function, I'd just add the if inside the body of the functional component:
```tsx
if (redirect) return <Redirect />
return (
  <>
    <Container />
  </>
)
```